Makefile

```diff
@@ -26,105 +26,109 @@
 # STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
 # THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+
+subbuilddir:=$(shell uname -o|sed -e 's|/|_|g')/$(shell git branch |grep '\* ' |cut -f2 -d' ')
+
+
 all: release-all
 
 cmake-debug:
-	mkdir -p build/debug
-	cd build/debug && cmake -D CMAKE_BUILD_TYPE=Debug ../..
+	mkdir -p build/$(subbuilddir)/debug
+	cd build/$(subbuilddir)/debug && cmake -D CMAKE_BUILD_TYPE=Debug ../../../..
```

Comment:
If you have a branch name in this, then you need to start quoting.

Reply:
And replacing / with something else (I do have a branch with a / in so it's not pedantic :P)

Reply:
ooops, never thought about / in a branch name :D 

Ok I will add some quote here and there and test with OMG branch name.(I just mod the Makefile for my own purpose and thought it could be useful for other with too slow machine :p)